network.go

```diff
@@ -68,9 +68,26 @@ const (
 	defaultQueues     = 8
 )
 
-type netIfaceAddrs struct {
-	iface net.Interface
-	addrs []net.Addr
+// NetIfaceAddrs describes a network interface an its associated IP addresses.
+type NetIfaceAddrs struct {
+	Iface net.Interface
```

Comment:
net.Interface gives you
```
// Interface represents a mapping between network interface name
  // and index. It also represents network interface facility
  // information.
  type Interface struct {
  	Index        int          // positive integer that starts at one, zero is never used
  	MTU          int          // maximum transmission unit
  	Name         string       // e.g., "en0", "lo0", "eth0.100"
  	HardwareAddr HardwareAddr // IEEE MAC-48, EUI-48 and EUI-64 form
  	Flags        Flags        // e.g., FlagUp, FlagLoopback, FlagMulticast
  }
  ```

however  netlink.Link.Attrs() gives you
```
// LinkAttrs represents data shared by most link types
type LinkAttrs struct {
	Index        int
	MTU          int
	TxQLen       int // Transmit Queue Length
	Name         string
	HardwareAddr net.HardwareAddr
	Flags        net.Flags
	RawFlags     uint32
	ParentIndex  int         // index of the parent link device
	MasterIndex  int         // must be the index of a bridge
	Namespace    interface{} // nil | NsPid | NsFd
	Alias        string
	Statistics   *LinkStatistics
	Promisc      int
	Xdp          *LinkXdp
	EncapType    string
	Protinfo     *Protinfo
	OperState    LinkOperState
	NetNsID      int
}
```

which is much more comprehensive. So how about switching to that? That will help longer term.